generators/paths/domain_parser/code_snippet_common.md

```diff
@@ -0,0 +1,36 @@
+```python
+import re
+
+def domain_parser(link:str, include_sub_domain:bool = False) -> str:
+    """
+    @param link: the input link
+    @param include_sub_domain: Attribute whether subdomain in front of domain is desired or not
+    @return: rootdomain
+    """
+
+    clean_link = re.sub("www.", "",link)
+    parts = clean_link.split("/")
+    domain = parts[2]
```

Comment:
e.g. "linkedin.com", "www.linkedin.com/mynetwork/epic" and "www.linkedin.com" are viable links for browsers. Not sure if we need to accommodate for that but the logic wouldn't work for them.

All three would be considered invalid URLs following the valid_url function we build for the URL keyword extractor. 

Also, (as long as the URLs are valid) the urlpaser might help prevent splitting URLs (netloc = domain for valid URLs)
`from urllib.parse import urlparse`
`print(urlparse(link))`


![image](https://github.com/code-kern-ai/bricks/assets/91723236/40343e1b-11e1-473c-98f8-9a30574e2b6c)